package-res/resources/web/pentaho/data/filter.js

```diff
@@ -0,0 +1,165 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+  "./filter/IsEqual",
+  "./filter/IsIn",
+  "./filter/And",
+  "./filter/Or",
+  "./filter/Not"
+], function(IsEqual, IsIn, AndFilter, OrFilter, NotFilter) {
+  "use strict";
+
+  /**
+   * @name RootFilter
+   * @memberOf pentaho.data.filter
+   * @extends pentaho.data.filter.Or
+   * @class
+   * @ignore
+   *
+   * @param {Object} filterSpec The specification of a Filter.
+   *
+   */
+  var RootFilter = OrFilter.extend("pentaho.data.RootFilter", /** @lends pentaho.data.RootFilter# */{
+    constructor: function(filterSpec) {
+      if(filterSpec)
+        this.base(fromSpec(filterSpec));
+      else
+        this.base();
+    }
+  });
+
+  /**
+   * This namespace contains classes for filters that are used for expressing a data set intensionally.
```

Comment:
Would be good to link [intensionally](https://en.wikipedia.org/wiki/Intensional_definition) to the intended meaning.